src/codegen/compile.ml

```diff
@@ -4488,7 +4491,8 @@ module Serialization = struct
           (* Second time we see this *)
           (* Calculate relative offset *)
           let (set_offset, get_offset) = new_local env "offset" in
-          get_tag ^^ get_data_buf ^^ G.i (Binary (Wasm.Values.I32 I32Op.Sub)) ^^
+          get_x ^^ Heap.load_field Tagged.tag_field ^^
```

Comment:
Should the initial load aboce `get_x ^^ Tagged.load ^^ set_tag` also load the full header and read `get_x ^^ Heap.load_field Tagged.tag_field`? 

Note that the offsets need no be even or aligned in any way, AFAIK, since these are just offsets into the blob, not aligned heap addresses, right? Serialized objects are in no way aligned on word boundaries.

Reply:
> Note that the offsets need no be even or aligned in any way, AFAIK, since these are just offsets into the blob, not aligned heap addresses, right? Serialized objects are in no way aligned on word boundaries.

Right, Joachim confirmed this in https://github.com/dfinity/motoko/pull/2793#issuecomment-923251202